kyo-core/shared/src/main/scala/kyo/Clock.scala

```diff
@@ -87,7 +100,18 @@ object Clock:
     val live: Clock =
         Clock(
             new Unsafe:
-                def now()(using AllowUnsafe): Instant = Instant.fromJava(java.time.Instant.now())
+                val executor                 = Executors.newScheduledThreadPool(2, Threads("kyo-core-clock-executor"))
+                def now()(using AllowUnsafe) = Instant.fromJava(java.time.Instant.now())
+                def sleep(duration: Duration) =
```

Comment:
I had to move the management of timed tasks to `Clock` so it's possible to customize the behavior for time shifting and control